fbw-a32nx/src/systems/instruments/src/EFB/Failures/FailureGenerators/AltitudeFailureGenerator.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useMemo } from 'react';
+import { useSimVar } from '@instruments/common/simVars';
+import {
+    activateRandomFailure, basicData, FailureGenContext, FailureGenData, failureGeneratorCommonFunction,
+    FailurePhases, flatten, setNewSetting,
+} from 'instruments/src/EFB/Failures/FailureGenerators/RandomFailureGen';
+import { usePersistentProperty } from '@instruments/common/persistence';
+import { t } from 'instruments/src/EFB/translation';
+import { findGeneratorFailures } from 'instruments/src/EFB/Failures/FailureGenerators/FailureSelection';
+import { ArrowDownRight, ArrowUpRight } from 'react-bootstrap-icons';
+import { ButtonIcon, FailureGeneratorChoiceSetting, FailureGeneratorSingleSetting } from 'instruments/src/EFB/Failures/FailureGenerators/FailureGeneratorSettingsUI';
+
+const settingName = 'EFB_FAILURE_GENERATOR_SETTING_ALTITUDE';
+const additionalSetting = [2, 1, 0, 80, 250];
+const numberOfSettingsPerGenerator = 5;
+const uniqueGenPrefix = 'A';
+const failureGeneratorArmed :boolean[] = [];
+const genName = 'Altitude';
+const alias = () => t('Failures.Generators.GenAlt');
+const disableTakeOffRearm = false;
+const rolledDice:number[] = [];
```

Comment:
I was hoping to have one file carrying the logic and settings (constants) of each specific failure generator's.
Apart from the Alias that is probably a bad idea there, why would that be an issue ?

Reply:
There are deeper concerns about whether this is following the expected pattern in react-js i.e. the flow from parent to child components but anyway this is looking like it would require an extensive refactor to correct. 

For now though, we still need input on the UI and whether this makes sense from a user perspective to begin with.